bundles/org.openhab.automation.jsscriptingnashorn/src/main/java/org/openhab/automation/jsscriptingnashorn/internal/NashornScriptEngineFactory.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jsscriptingnashorn.internal;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.module.script.AbstractScriptEngineFactory;
+import org.openhab.core.automation.module.script.ScriptEngineFactory;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * This is an implementation of a {@link ScriptEngineFactory} for Nashorn.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+@Component(service = ScriptEngineFactory.class)
+@NonNullByDefault
+public class NashornScriptEngineFactory extends AbstractScriptEngineFactory {
+
+    private static final String NASHORN_ARGS = "nashorn.args";
+
+    private final org.openjdk.nashorn.api.scripting.NashornScriptEngineFactory factory = new org.openjdk.nashorn.api.scripting.NashornScriptEngineFactory();
+
+    private final List<String> scriptTypes = (List<String>) Stream.of(factory.getExtensions(), factory.getMimeTypes())
+            .flatMap(List::stream) //
+            .collect(Collectors.toUnmodifiableList());
+
+    public NashornScriptEngineFactory() {
+        removeUnsupportedNashornArgs();
+    }
+
+    private void removeUnsupportedNashornArgs() {
+        String property = System.getProperty(NASHORN_ARGS, "");
+        property = property.replaceAll("--no-deprecation-warning", "").trim();
+        System.setProperty(NASHORN_ARGS, property);
+    }
+
```

Comment:
```suggestion
```

This can be removed once the property is removed from [setenv](https://github.com/openhab/openhab-distro/blob/466a3427711fe90b4b6511808b461edcc6ed6b12/distributions/openhab/src/main/resources/bin/setenv#L109) / [setenv.bat](https://github.com/openhab/openhab-distro/blob/466a3427711fe90b4b6511808b461edcc6ed6b12/distributions/openhab/src/main/resources/bin/setenv.bat#L126). The default Nashorn engine will no longer warn about its deprecation because it has been removed. :wink: